onnxruntime/core/providers/webnn/builders/impl/split_op_builder.cc

```diff
@@ -0,0 +1,162 @@
+    } else {
+      // w/o 'split' input for opset 13
+      // split: Optional length of each output. Values should be >= 0.Sum of the values must be equal to the dim value at 'axis' specified.
+      output_array = model_builder.GetBuilder().call<emscripten::val>("split", input, static_cast<int32_t>(1), options);
+      ORT_RETURN_IF_NOT(output_array["length"].as<int32_t>() == static_cast<int32_t>(1), "The size of outputs must be equal to 1.");
+    }
+  }
+  for (int64_t i = 0; i < output_array["length"].as<int32_t>(); i++) {
+    model_builder.AddOperand(node.OutputDefs()[i]->Name(), std::move(output_array[i]));
+  }
+  return Status::OK();
+}
+
+// Operator support related.
+
+int SplitOpBuilder::GetMinSupportedOpSet(const Node& /* node */) const {
+  // Since opset 13, Split has optional 'split' input.
+  return 13;
+}
+
+bool SplitOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& initializers, const Node& node,
+                                       const logging::Logger& logger) const {
+  const auto& input_defs = node.InputDefs();
+  std::vector<int64_t> input_shape;
+  if (!GetShape(*input_defs[0], input_shape, logger)) {
+    LOGS(logger, VERBOSE) << "Cannot get input's shape.";
+    return false;
+  }
+  const auto rank = input_shape.size();
+
+  NodeAttrHelper helper(node);
+  int64_t axis = helper.Get("axis", 0);
+  axis = HandleNegativeAxis(axis, rank);
+
+  if (input_defs.size() == 2) {
+    // Inputs contains optional 'split' input
+    const auto& split_name = input_defs[1]->Name();
+    if (!Contains(initializers, split_name)) {
+      LOGS(logger, VERBOSE) << "The split must be a constant initializer.";
+      return false;
+    }
+    // Values should be >= 0.Sum of the values must be equal to the dim value at 'axis' specified.
+    std::vector<int64_t> split;
+    const auto& split_tensor = *initializers.at(input_defs[1]->Name());
+    if (!GetShapeByTensor(split_tensor, split, logger)) {
+      LOGS(logger, VERBOSE) << "Cannot get split.";
+      return false;
+    }
+    int64_t sum = 0;
+    for (int64_t i = 0; i < split.size(); i++) {
+      if (split[i] < 0) {
+        LOGS(logger, VERBOSE) << "Value of split should be greater than or equal to 0.";
+        return false;
+      }
+      sum += split[i];
+    }
+    if (sum != input_shape[axis]) {
+      LOGS(logger, VERBOSE) << "Sum of the split's values must be equal to the dim value at 'axis' specified.";
+      return false;
+    }
+  } else if (input_defs.size() == 1) {
+    if (helper.HasAttr("num_outputs")) {
+      // Split has 'num_outputs' attribute when opset is 18.
+      const int32_t num_outputs = helper.Get("num_outputs", 1);
+      if (num_outputs < 1) {
+        LOGS(logger, VERBOSE) << "The 'num_outputs' must be a positive integer.";
+        return false;
+      }
+    } else {
+      const auto opset = node.SinceVersion();
+      if (opset == 18) {
```

Comment:
```suggestion
      if (opset >= 18) {
```

Just to make it slightly more future proof (because many times new versions of ops are registered that are functionally identical, and they only add more data types like bfloat16 or float8...).